lib/ipmi_sel.c

```diff
@@ -703,6 +703,82 @@ get_supermicro_evt_desc(struct ipmi_intf *intf, struct sel_event_record *rec)
 	return desc;
 }
 
+char *
+get_quanta_evt_desc(struct ipmi_intf *intf, struct sel_event_record *rec)
+{
+	struct ipmi_rs *rsp;
+	struct ipmi_rq req;
+	char *desc = NULL;
+	int chipset_type = 0;
+	int data1;
+	int data2;
+	int data3;
+	int sensor_type;
+	uint8_t i = 0;
+	uint16_t oem_id = 0;
+	/* Get the OEM event Bytes of the SEL Records byte 13, 14, 15 to
+	 * data1,data2,data3
+	 */
+	data1 = rec->sel_type.standard_type.event_data[0];
+	data2 = rec->sel_type.standard_type.event_data[1];
+	data3 = rec->sel_type.standard_type.event_data[2];
+	/* Check for the Standard Event type == 0x6F */
+	if (rec->sel_type.standard_type.event_type != 0x6F) {
+		return NULL;
+	}
+	/* Allocate mem for te Description string */
+	desc = malloc(sizeof(char) * SIZE_OF_DESC);
+	if (desc == NULL) {
+		lprintf(LOG_ERR, "ipmitool: malloc failure");
+		return NULL;
+	}
+	memset(desc, '\0', SIZE_OF_DESC);
+	sensor_type = rec->sel_type.standard_type.sensor_type;
+	switch (sensor_type) {
+		case SENSOR_TYPE_MEMORY:
+			memset(&req, 0, sizeof (req));
+			req.msg.netfn = IPMI_NETFN_APP;
+			req.msg.lun = 0;
+			req.msg.cmd = BMC_GET_DEVICE_ID;
+			req.msg.data = NULL;
+			req.msg.data_len = 0;
+
+			rsp = intf->sendrecv(intf, &req);
+			if (rsp == NULL) {
+				lprintf(LOG_ERR, " Error getting system info");
+				if (desc != NULL) {
+					free(desc);
+					desc = NULL;
+				}
+				return NULL;
+			} else if (rsp->ccode > 0) {
+				lprintf(LOG_ERR, " Error getting system info: %s",
+						val2str(rsp->ccode, completion_code_vals));
+				if (desc != NULL) {
+					free(desc);
+					desc = NULL;
+				}
+				return NULL;
+			}
+			/* check the platform type */
+			oem_id = ipmi_get_qct_paltform_id(intf);
+						
+			if(oem_id == 0x02)
+			{
+				snprintf(desc, SIZE_OF_DESC, "CPU%d_%c%d",
+						 (data3 & 0xC0)>> 6,
+						 ((data3 & 0x38) >> 3)+0x41,						 
+						 (data3 & 0x07));
```

Comment:
Please indent/align according to https://github.com/ipmitool/ipmitool/wiki/Coding-Standards